### PR TITLE
Add check for starboard python module

### PR DESCRIPTION
**Pull request by oxve**
b/299098707

### CODE REVIEW

File: build/util/has_starboard.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script for checking if we have the starboard python modules on the path."""
+
+try:
+  import starboard # pylint: disable=unused-import
```
Comment (andrewsavage1) [inline build/util/has_starboard.py:18]: Wouldn't this incorrectly pass if PYTHONPATH is set to point to the internal repo but we're building the external repo or vice versa?

Comment (oxve) [reply build/util/has_starboard.py:18]: Indeed it would. I guess it should use the same check as was in `gn.py`

Comment (arjungm) [reply build/util/has_starboard.py:18]: I'm planning to abandon this change: https://github.com/youtube/cobalt/pull/1531/files

but it'd be good to ensure the path-checks are robust for windows quirkiness with forward and backslash

**Review by arjungm — CHANGES_REQUESTED** [review]

File: build/util/has_starboard.py
Comment (andrewsavage1) [inline build/util/has_starboard.py:1]: I think this file needs to be renamed now

File: build/util/has_starboard.py (hunk 1 above)
Comment (oxve) [reply build/util/has_starboard.py]: I realized that the changes in #1531 doesn't handle a PYTHONPATH with more than one entry. I fixed that in my script and tested it with a multitude of variations.

File: build/util/has_starboard.py
Comment (oxve) [reply build/util/has_starboard.py:1]: Renamed to `build/util/is_on_path.py`

File: build/util/is_on_path.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import os
+import sys
+
+SRC_DIR = os.path.abspath(os.path.join(
+    os.path.dirname(__file__), os.pardir, os.pardir))
+
+def main():
+  abs_path_to_me = os.path.abspath(__file__)
+  rel_path_to_me = os.path.relpath(__file__, start=SRC_DIR)
```
Comment (andrewsavage1) [inline build/util/is_on_path.py:25]: Can't we just check if SRC_DIR is in sys.path?

Comment (oxve) [reply build/util/is_on_path.py:24]: No, we need to check specifically for the file to see if another cobalt repo is on the path in front of this one. I could theoretically look for any file in the cobalt repo but the check must be done nonetheless.

I could see `SRC_DIR` is _the_ first path on `PYTHONPATH` but that feels a little limiting.

Comment (andrewsavage1) [reply build/util/is_on_path.py:24]: Got it

**Review by andrewsavage1 — APPROVED** [review]

**Review by arjungm — APPROVED** [review]

File: build/util/is_on_path.py
Comment (andrewsavage1) [inline build/util/is_on_path.py:1]: Sorry, one last thing, would prefer if this was under starboard/build instead of build/ (which is owned by upstream)

Comment (oxve) [reply build/util/is_on_path.py:1]: Sg, done

**Review by andrewsavage1 — APPROVED** [review]

**Pull request merged**